R/methods.R

```diff
@@ -88,35 +88,206 @@ plot.ontram_history <- function(object, col_train = "blue", col_test = "red", ..
          ylab = "negative logLik", ... = ...)
   }
   lines(epoch, object$test_loss, col = col_test)
+  if (add_best) {
+    if (is.null(object$epoch_best)) stop("`epoch_best` not found.")
+    abline(v = object$epoch_best, lty = 2)
+  }
   legend("topright", legend = c("Train", "Test"), col = c(col_train, col_test),
          bty = "n", lwd = 2)
 }
 
-# Function for saving ontram models
-save_model.ontram <- function(object, filename, ...) {
-  nm_theta <- paste0(filename, "_theta.h5")
-  nm_beta <- paste0(filename, "_beta.h5")
-  nm_rest <- paste0(filename, "_r.Rds")
-  rest <- list(x_dim = object$x_dim,
-               y_dim = object$y_dim,
-               n_batches = object$n_batches,
-               epochs = object$epochs)
-  save(rest, file = nm_rest)
-  save_model_hdf5(object$mod_baseline, nm_theta)
-  save_model_hdf5(object$mod_shift, nm_beta)
+#' Simulate Responses
+#' @method simulate ontram
+#' @examples
+#' data("wine", package = "ordinal")
+#' fml <- rating ~ temp + contact
+#' x_train <- model.matrix(fml, data = wine)[, -1L]
+#' y_train <- model.matrix(~ 0 + rating, data = wine)
+#' x_valid <- x_train[1:20,]
+#' y_valid <- y_train[1:20,]
+#' mo <- ontram_polr(x_dim = ncol(x_train), y_dim = ncol(y_train),
+#'                   method = "logit", n_batches = 10, epochs = 50)
+#' fit_ontram(mo, x_train = x_train, y_train = y_train)
+#' simulate(mo, nsim = 1, x = x_valid, y = y_valid)
+#' @export
+simulate.ontram <- function(object, nsim = 1, seed = NULL, x = NULL, y, im = NULL, ...) {
+  if (!exists(".Random.seed", envir = .GlobalEnv, inherits = FALSE))
+    runif(1)
```

Comment:
This `runif(1)` call looks a bit weird. I'd probably assign it to a variable? Is it just to get the `".Random.seed"` initialized?